storage/innobase/gis/gis0sea.cc

```diff
@@ -289,10 +288,6 @@ rtr_pcur_getnext_from_path(
 			mtr->rollback_to_savepoint(1);
 		}
 
-		ut_ad((my_latch_mode | 4) == BTR_CONT_MODIFY_TREE
-		      || !page_is_leaf(btr_cur_get_page(btr_cur))
-		      || !btr_cur->page_cur.block->page.lock.have_any());
-
```

Comment:
We release all the page latches (mtr->rollback_to_savepoint(1)) when rtr_pcur_getnext_from_path() is called. This looks fine as we are either holding X (pessimistic) or SX(optimistic) latch at index level and have all the possible page IDs cached already.

Only issue is that cursor now points to a possible invalid block "btr_cur->page_cur.block". We eventually set the cursor to the next candidate page after latching it and there doesn't seem to be any real issue other than this debug assert trying to access the block.

The block can be freed even can be reused concurrently and MSAN looks to be catching it when it is indeed freed. We should not access the block/page at this point and It justifies the removal of the assert.

Can you please add a comment here stating the page cursor could point to invalid block at this point and should not be accessed ? A braver approach would be to set the page cursor block to nullptr immediately after rolling back to savepoint and assert here that the page cursor block is nullptr at this point.